doc/source/reference/simd/how-it-works.rst

```diff
@@ -0,0 +1,349 @@
+*****************************
+How the CPU dispatcher works?
+*****************************
+
+NumPy dispatcher is based on multi-source compiling, which means taking
+a certain source and compiling it multiple times with different compiler
+flags and also with different **C** definitions that affect the code
+paths to enable certain instruction-sets for each compiled object
+depending on the required optimizations, then combining the returned
+objects together.
```

Comment:
This is a very long sentence - might be re-worded, maybe something like:
```suggestion
NumPy dispatcher is based on multi-source compiling, which means taking
a certain source and compiling it multiple times with different compiler
flags and also with different **C** definitions that affect the code
paths. This enables certain instruction-sets for each compiled object
depending on the required optimizations which are then combined in the returned
objects together.
```

Reply:
Done, but I have changed the last phrase to become "and ends with linking the
returned objects together.".